.teamcity/src/main/kotlin/model/FunctionalTestBucketProvider.kt

```diff
@@ -86,10 +86,33 @@ class StatisticBasedFunctionalTestBucketProvider(val model: CIBuildModel, testBu
             val buckets: List<BuildTypeBucket> = testCoverageAndBucket.getJSONArray("buckets").map {
                 fromJsonObject(it as JSONObject).toBuildTypeBucket(model.subprojects)
             }
-            testCoverage to buckets
+
+            // Sometimes people may add new subproject into `subprojects.json`
+            // in this case we have no historical test running time, so we simply add these subprojects into first available bucket
+            val allSubprojectsInBucketJson = buckets.flatMap {
+                if (it is SmallSubprojectBucket) it.subprojects.map { it.name }
+                else listOf((it as LargeSubprojectSplitBucket).subproject.name)
+            }.toSet()
+            val allSubprojectsInModel = model.subprojects.subprojects.filter { it.functionalTests || it.unitTests || it.crossVersionTests }.map { it.name }.toMutableList()
```

Comment:
I suppose we don't want to use the full test coverage here, but only the one we are currently looking at, right? So check here something like
```
it.hasTestOf(testCoverage.testType)
```

Reply:
Done.